scarb/src/core/manifest/toml_manifest.rs

```diff
@@ -814,6 +838,18 @@ impl TomlManifest {
     }
 }
 
+/// Returns the absolute canonical path of the LICENSE file for a [`TomlPackage`].
+pub fn license_for_package(
+    package_root: &Utf8Path,
+    license_file: Option<&Utf8PathBuf>,
+) -> Result<Option<Utf8PathBuf>> {
+    if let Some(p) = license_file {
+        Ok(abs_canonical_path(package_root, Some(p.as_path()))?)
+    } else {
+        Ok(None)
+    }
+}
```

Comment:
Isn't this equivalent to:
```
abs_canonical_path(package_root, license_file.map(|p| p.as_path())
```
?

Maybe wrapping second argument of abs_canonical_path with AsRef would help?